sirepo/sim_api/jupyterhublogin.py

```diff
@@ -40,76 +44,35 @@ async def api_checkAuthJupyterHub(self):
-            if not u:
-                create_user(self)
-            return self.reply_redirect("jupyterHub")
-        return self.reply_ok()
+        if not _unchecked_jupyterhub_user_name(self):
+            create_user(self)
+        return self.reply_redirect("jupyterHub")
 
 
 def cfg():
     return _init()
 
 
-def create_user(qcall, github_handle=None, check_dir=False):
-    """Create a Jupyter user and possibly migrate their data from old jupyter.
-
-    Terminology:
-      migration user: A user with data at the old jupyter
-      jupyter user: The user of new jupyter
-
-    A few interesting cases to keep in mind:
-      1. If a user is migrating (has a github handle) we should never modify
-         the handle and if they are able to migrate then their username should be
-         their github handle (downcased).
-      2. User signs into sirepo under one@any.com. They migrate their data using
-         GitHub handle y. They sign into sirepo under two@any.com. They choose
-         to migrate GitHub handle y again. We should let them know that they
-         have already migrated.
-      3. one@any.com signs up for jupyter and does not migrate data. They are
-         given the username one. two@any.com signs up for jupyter and they
-         migrate their data, but they have no data to migrate. They have the
-         github handle one and no previous data. They should be alerted that
-         they can't migrate that GitHub handle.
-      4. A new user signs in with foo@any.com and they do not select to
-         migrate. There is an existing foo migration user which has not registered
-         yet. We should uniquify the new user (foo_xyz) to ensure the name
-         doesn't collide with the existing (yet to register) user.
-
-    Args:
-        github_handle (str): The user's github handle
-        check_dir (bool): assert that an existing user does not have a dir with
-                          the same name (before modifying the name to eliminate
-                          conflicts)
+def create_user(qcall):
+    """Create a Jupyter user if not one associated with Sirepo uid
+
+    New user_name is created from sanitized Sirepo username.
+    If user_name is taken, will be appended with random string.
+
     Returns:
-        user_name (str): The user_name of the new user
+        user_name (str): The user_name of the new or existing user
     """
 
-    def __handle_or_name_sanitized():
-        return re.sub(
-            r"\W+",
-            _HUB_USER_SEP,
-            # Get the local part of the email. Or in the case of another auth
-            # method (ex github) it won't have an '@' so it will just be their
-            # user name, handle, etc.
-            (github_handle or qcall.auth.logged_in_user_name()).split("@")[0],
-        ).lower()
+    def __name_sanitized():
+        return _INVALID_USER_NAME_CHARS.sub(
+            _HUB_USER_SEP, qcall.auth.local_part_user_name().lower()
```

Comment:
local_part_user_name should guarantee lower. A logged in user's email should always be in lower case.

Reply:
[3a9f0d1](https://github.com/radiasoft/sirepo/pull/7027/commits/3a9f0d1399801354cea9e77a661893218636e2dd)